maps/src/MapBinner.cxx

```diff
@@ -206,9 +220,8 @@ MapBinner::Process(G3FramePtr frame, std::deque<G3FramePtr> &out)
 		}
 
 		if (map_weights_) {
-			out_frame->Put((Q_) ? "Wpol" : "Wunpol", map_weights_);
-			map_weights_ = G3SkyMapWeightsPtr(new G3SkyMapWeights(
-			    T_, T_->GetPolConv() != G3SkyMap::ConvNone));
+			out_frame->Put(pol_ ? "Wpol" : "Wunpol", map_weights_);
```

Comment:
Why not just clone the old `map_weights_` here like the T code does? Then you don't have to maintain any parallel state.